core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -444,13 +444,12 @@ class UnifiedLog(@volatile var logStartOffset: Long,
    */
   def maybeUpdateHighWatermark(hw: Long): Option[Long] = {
     lock.synchronized {
-      val oldHighWatermark = highWatermarkMetadata
-      updateHighWatermark(LogOffsetMetadata(hw)) match {
-        case oldHighWatermark.messageOffset =>
-          None
-        case newHighWatermark =>
-          Some(newHighWatermark)
-      }
+      val oldHighWatermark = highWatermarkMetadata.messageOffset()
+      val updatedHighWatermark = updateHighWatermark(new LogOffsetMetadata(hw))
+      if (updatedHighWatermark == oldHighWatermark)
+        None
+      else
+        Some(updatedHighWatermark)
```

Comment:
Is this change actually required?

Reply:
I was required previously as we can't have getters as an `case` option. Now that we're moving to public fields this change is not needed anymore, so I'll revert it.